crypto/noise/src/cipher_state.rs

```diff
@@ -161,12 +161,19 @@ impl<Cipher: NoiseCipher> CipherState<Cipher> {
         self.cipher.is_some()
     }
 
+    /// Retrieve the nonce value which will be used in the next operation.
+    ///
+    /// This is an extension of the noise protocol to allow for implicit-nonce
+    /// writers with explicit-nonce readers to co-exist in the same stream.
+    pub fn next_nonce(&self) -> u64 {
+        self.nonce
+    }
+
     /// The noise protocol `SetNonce()` operation.
     ///
     /// This will irrevocably override the current nonce value.
     pub fn set_nonce(&mut self, nonce: u64) {
```

Comment:
Where does this get called now? Perhaps this is implemented in a follow-up PR?

Reply:
This was already called from the `Ready::decrypt_with_nonce()` method.